service/src/main/java/bio/terra/workspace/app/controller/ControlledGcpResourceApiController.java

```diff
@@ -537,51 +551,111 @@ private ApiCloneControlledGcpBigQueryDatasetResult fetchCloneBigQueryDatasetResu
+   * The only time it is allowed is for application-private resources. If it is populated, we
+   * validate the user email and the specified IAM roles.
+   *
+   * <p>If this is a user-private resource, then we compute the default to be the user email of the
+   * userRequest and give it the default role.
    *
-   * <p>Shared access resources must not specify private resource roles. Private access resources
-   * must specify at least one private resource role.
+   * @param commonFields common fields from a controlled resource create request
+   * @param userRequest authenticate user
+   * @param defaultUserRole for user private resources, the default role to grant
+   * @return PrivateUserRole holding the user email and the role list
    */
-  private List<ControlledResourceIamRole> privateRolesFromBody(
-      ApiControlledResourceCommonFields commonFields) {
-    List<ControlledResourceIamRole> privateRoles =
-        Optional.ofNullable(commonFields.getPrivateResourceUser())
-            .map(
-                user ->
-                    user.getPrivateResourceIamRoles().stream()
-                        .map(ControlledResourceIamRole::fromApiModel)
-                        .collect(Collectors.toList()))
-            .orElse(Collections.emptyList());
-    // Validate that we get the private role when the resource is private and do not get it
-    // when the resource is public
+  private PrivateUserRole validatePrivateResourceUser(
+      UUID workspaceId,
+      ApiControlledResourceCommonFields commonFields,
+      AuthenticatedUserRequest userRequest,
+      ControlledResourceIamRole defaultUserRole) {
+
     AccessScopeType accessScope = AccessScopeType.fromApi(commonFields.getAccessScope());
-    if (accessScope == AccessScopeType.ACCESS_SCOPE_PRIVATE && privateRoles.isEmpty()) {
-      throw new ValidationException("At least one IAM role is required for private resources");
+    ManagedByType managedBy = ManagedByType.fromApi(commonFields.getManagedBy());
+    ApiPrivateResourceUser inputUser = commonFields.getPrivateResourceUser();
+
+    // Shared access has no private user role
+    if (accessScope == AccessScopeType.ACCESS_SCOPE_SHARED) {
+      validateNoInputUser(inputUser);
+      return new PrivateUserRole(false, null, null);
+    }
+
+    // Private access scope
+    switch (managedBy) {
+      case MANAGED_BY_APPLICATION:
+        {
+          // Supplying a user is optional for applications
+          if (inputUser == null) {
+            return new PrivateUserRole(false, null, null);
+          }
+
+          // We have a private user, so make sure the email is present and valid
+          String userEmail = commonFields.getPrivateResourceUser().getUserName();
+          ControllerValidationUtils.validateEmail(userEmail);
+
+          /*
+           TODO: the APPLICATION role does not have read_policies action, so cannot perform
```

Comment:
Looks like it's there now? https://github.com/broadinstitute/sam/blob/develop/src/main/resources/reference.conf#L134

Reply:
As of a late commit yesterday. By Monday when I finish reworking this code, I'll be able to enable it!

Reply:
Removed and working!